[BUG] Unable to build the project from sources on macOS

**************************
### Code of Conduct
_By submitting this report you automatically agree that you've read and accepted the following conditions._ 
- Support for DQ Robotics is given voluntarily and it's not the developers' role to educate and/or convince anyone of their vision.
- Any possible response and its timeliness will be based on the relevance, accuracy, and politeness of a request and the following discussion.
- If a DQ Robotics member replies, the user must let them know if their response solves their issue or not. 
- Any suggestion/advice/request made by anyone, as well intentioned as they might be, might not be incorporated into DQ Robotics.
*************************

**Bug description**

Hi @mmmarinho, 

I am having some issues building the project from sources on macOS. I think the problem started after the commit 2892989897c26e1539aea2b8ccd3076a403f1a11, in which the file `dependencies.cmake` was removed. Specifically, the `dependencies.cmake` file used to include the following lines:

```cmake
if(APPLE) #APPLE
    INCLUDE_DIRECTORIES(
        /usr/local/include/
        /usr/local/include/eigen3
        # Most recent versions of brew install here
        /opt/homebrew/include
        /opt/homebrew/include/eigen3
    )
    ADD_COMPILE_OPTIONS(-Werror=return-type -Wall -Wextra -Wmissing-declarations -Wredundant-decls -Woverloaded-virtual)
    # The library is installed here when using the regular cmake ., make, sudo make install
    LINK_DIRECTORIES(
        /usr/local/lib/
        /opt/homebrew/lib/
        )

    find_package(cppzmq REQUIRED)
    find_package(ZeroMQ REQUIRED)

endif()
```
(By manually adding the aforementioned lines in the CMakeLists.txt, the compilation works on macOS.)

OS: macOS 15.5

```
mkdir build
cd build/
cmake ..
make
sudo make install
```
*Output*
```
-- The CXX compiler identification is AppleClang 17.0.0.17000013
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Eigen3: /opt/homebrew/include/eigen3 (Required is at least version "2.91.0")
-- Found PkgConfig: /opt/homebrew/bin/pkg-config (found version "2.3.0")
-- Found ZeroMQ: /opt/homebrew/lib/libzmq.dylib
-- Configuring done (8.1s)
-- Generating done (0.0s)
-- Build files have been written to: /Users/user/git/dqrobotics/dqrobotics-interface-coppeliasim/build
[ 20%] Building CXX object CMakeFiles/dqrobotics-interface-coppeliasim-zmq.dir/src/dqrobotics/interfaces/coppeliasim/internal/_zmq_wrapper.cpp.o
[ 40%] Building CXX object CMakeFiles/dqrobotics-interface-coppeliasim-zmq.dir/src/dqrobotics/interfaces/coppeliasim/DQ_CoppeliaSimInterfaceZMQ.cpp.o
/Users/user/git/dqrobotics/dqrobotics-interface-coppeliasim/src/dqrobotics/interfaces/coppeliasim/DQ_CoppeliaSimInterfaceZMQ.cpp:27:10: fatal error: 'dqrobotics/DQ.h' file not found
   27 | #include "dqrobotics/DQ.h"
      |          ^~~~~~~~~~~~~~~~~
1 error generated.
make[2]: *** [CMakeFiles/dqrobotics-interface-coppeliasim-zmq.dir/src/dqrobotics/interfaces/coppeliasim/DQ_CoppeliaSimInterfaceZMQ.cpp.o] Error 1
make[1]: *** [CMakeFiles/dqrobotics-interface-coppeliasim-zmq.dir/all] Error 2
make: *** [all] Error 2
[ 20%] Building CXX object CMakeFiles/dqrobotics-interface-coppeliasim-zmq.dir/src/dqrobotics/interfaces/coppeliasim/DQ_CoppeliaSimInterfaceZMQ.cpp.o
/Users/user/git/dqrobotics/dqrobotics-interface-coppeliasim/src/dqrobotics/interfaces/coppeliasim/DQ_CoppeliaSimInterfaceZMQ.cpp:27:10: fatal error: 'dqrobotics/DQ.h' file not found
   27 | #include "dqrobotics/DQ.h"
      |          ^~~~~~~~~~~~~~~~~
1 error generated.
make[2]: *** [CMakeFiles/dqrobotics-interface-coppeliasim-zmq.dir/src/dqrobotics/interfaces/coppeliasim/DQ_CoppeliaSimInterfaceZMQ.cpp.o] Error 1
make[1]: *** [CMakeFiles/dqrobotics-interface-coppeliasim-zmq.dir/all] Error 2
make: *** [all] Error 2
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -10,6 +10,8 @@ set(CMAKE_CXX_STANDARD_REQUIRED ON)
 find_package(Eigen3 REQUIRED)
 if (APPLE)
     find_package(cppzmq REQUIRED)
+    INCLUDE_DIRECTORIES(/usr/local/include/)
+    LINK_DIRECTORIES(/usr/local/lib/)
 endif()
 
 add_library(${PROJECT_NAME} 

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -10,6 +10,12 @@ set(CMAKE_CXX_STANDARD_REQUIRED ON)
 find_package(Eigen3 REQUIRED)
 if (APPLE)
     find_package(cppzmq REQUIRED)
+
+    # The DQ Robotics library is installed in /usr/local/include/ on macOS platforms.
+    # The following directives are required to find and link the DQ Robotics to the
+    # dqrobotics-interface-coppeliasim-zmq. Otherwise, this project could not compile
+    # from sources on macOS, as reported in
+    # https://github.com/dqrobotics/cpp-interface-coppeliasim-zmq/issues/16
     INCLUDE_DIRECTORIES(/usr/local/include/)
     LINK_DIRECTORIES(/usr/local/lib/)
 endif()

